modules/mvt/src/lib/parse-mvt.js

```diff
@@ -14,19 +14,27 @@ export default function parseMVT(input, options) {
   }
 
   const tile = new VectorTile(new Protobuf(input));
-  const mvtOptions = options.mvt;
+  const loaderOptions = options.mvt;
   const features = [];
 
-  for (const layerName in tile.layers) {
+  const selectedLayers = Array.isArray(loaderOptions.layers)
+    ? loaderOptions.layers
+    : Object.keys(tile.layers);
+
+  selectedLayers.forEach(layerName => {
     const vectorTileLayer = tile.layers[layerName];
 
+    if (!vectorTileLayer) {
+      return;
+    }
+
     for (let i = 0; i < vectorTileLayer.length; i++) {
       const vectorTileFeature = vectorTileLayer.feature(i);
 
-      const decodedFeature = getDecodedFeature(vectorTileFeature, mvtOptions);
+      const decodedFeature = getDecodedFeature(vectorTileFeature, {...loaderOptions, layerName});
```

Comment:
Perf: construct this object outside of the for loop

Reply:
Oh sorry! :( Didn't notice of this!